src/lib/Autodiff.hs

```diff
@@ -133,11 +133,14 @@ linearizeOp op = case op of
   FstRef   ref           -> (FstRef   <$> la ref           ) `bindLin` emitOp
   SndRef   ref           -> (SndRef   <$> la ref           ) `bindLin` emitOp
   Select p t f           -> (Select p <$> la t   <*> la f  ) `bindLin` emitOp
-  PtrLoad _              -> emitWithZero -- XXX: This assumes that pointers are always constants
+  -- XXX: This assumes that pointers are always constants
```

Comment:
That... Seems like a very strong assumption once we have `PtrStore`. Maybe we should allow `PtrLoad`, but disallow storing?

I wonder if those cases are ever even necessary. Maybe they should all be errors? I don't think we should end up in a situation where we have active pointers.